core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -73,8 +73,7 @@ public OutputFile newOutputFile(String path) {
   @Override
   public void deleteFile(String path) {
     Path toDelete = new Path(path);
-    FileSystem fs = Util.getFs(toDelete, hadoopConf.get());
-    try {
+    try (FileSystem fs = Util.getFs(toDelete, hadoopConf.get())) {
```

Comment:
This will remove the FileSystem from the cache.
If we delete multiple files on S3, closing the FS every time will become costly